### ISSUE TITLE
Duplicate variable passing to a module produces duplicate values

### ISSUE THREAD

**Issue by jurajseffer** [hashicorp/terraform#13283]
### Terraform Version
`Terraform v0.8.4`

### Affected Resource(s)
Modules

### Terraform Configuration Files
```hcl
module "test" {
  source = "./test"
  ips = ["1.2.3.4/32"]
  ips = ["1.2.3.4/32"]
}
```

### Expected Behavior
Warning/error is issued that a variable has been defined/passed twice.

### Actual Behavior
Values are silently merged into an array resulting in duplicate values, i.e. variable `ips` in test module has two entries.

**Comment by joshuaspence** [comment]
Similar but different to #13280

**Comment by apparentlymart** [comment]
Hi @jurajseffer! Thanks for reporting this.

As @joshuaspence noted, this is a similar problem to #13280 although it was manifesting in a different way here due to differences in how Terraform v0.12 dealt with module variables vs. resource arguments.

The fix for both is the same though, and so my validation of the fix in #13280 applies here too... in all contexts, it is invalid to assign the same attribute twice in the same block.

Thanks again for reporting this, and sorry for the delay in getting it fixed.
